utils/reporter_project_query.py

```diff
@@ -0,0 +1,158 @@
+import json
+import argparse
+import requests
+import numpy as np
+import math
+import pandas as pd
+
+
+def get_args():
+
+    parser = argparse.ArgumentParser(
+        description="Access and validate tables from Synapse"
+    )
+    parser.add_argument(
+        "-g",
+        nargs="+",
+        help="Space separated list of grant numbers to query NIH RePORTER",
+    )
+    parser.add_argument(
+        "-y",
+        nargs="+",
+        default=[],
+        help="The list of years to use when searching for grant information.",
+    )
+    parser.add_argument("-c", help="Path for CSV where converted JSON will be stored.")
+
+    return parser.parse_args()
+
+
+def build_payload(grant_numbers, years, lim):
+
+    criteria_dict = dict(
+        project_nums=grant_numbers, sub_project_only=True, fiscal_years=years
+    )
+
+    include = [
+        "ProjectTitle",
+        "ProjectNum",
+        "SubprojectId",
+        "FiscalYear",
+        "ProjectEndDate",
+    ]
+
+    payload = dict(
+        criteria=criteria_dict,
+        include_fields=include,
+        offset=0,
+        limit=lim,
+        sort_field="fiscal_year",
+        sort_order="desc",
+    )
+
+    json_payload = json.dumps(payload)
+
+    return json_payload
+
+
+def get_reporter_info(search_payload, header_content):
+
+    req = requests.post(
+        url="https://api.reporter.nih.gov/v2/projects/search",
+        headers=header_content,
+        data=search_payload,
+    )
+
+    return req
+
+
+def build_report(report_list, out_path):
+
+    df_list = []
+    for report in report_list:
+        report_df = pd.read_json(report, orient="records")
+        df_list.append(report_df)
+
+    full_report = pd.concat(df_list).reset_index(drop=True)
+
+    print(f"\n\nYour report is available at {out_path}.\n\n")
+
+    full_report.to_csv(out_path, index=False)
+
+
+def main():
+
+    args = get_args()
+
+    in_grants, in_years, csvPath = args.g, args.y, args.c
+
+    grants = []
+
+    years = []
+
+    for grant in in_grants:
+        grant = "".join([grant, "*"])
+        grants.append(grant)
+
+    for year in in_years:
+        year = int(year)
+        years.append(year)
+
+    headers = {"content-type": "application/json"}
+
+    max_request_length = int(400)
+    grant_count = int(len(grants))
+    req_per_grant = int(20)
```

Comment:
What's the reason for the `int` type castings?

Reply:
That's a good question! I don't 100% remember, but I think I added the `int` casting to address an error from the `/` operation on line 108, which was complaining about not being able to divide `str` objects